File: src/windows/Maple.WinUI/MapleUI/Templates/MuiTabShellMath.cs
namespace Maple.UI
{
    /// <summary>Tab bar placement for <see cref="MuiTabShell"/>. Declared
    /// here (not in the WinUI-dependent <see cref="MuiTabShell"/> control
    /// file) so this WinUI-free enum + its placement math can be linked
    /// into Maple.WinUI.Tests without pulling in WinUI — same split as
    /// <see cref="MuiDialogResultKind"/>/<see cref="MuiDialogLogic"/>.
    /// </summary>
    public enum MuiTabShellPlacement { Auto, Top, Bottom }

    /// <summary>
    /// Pure tab-bar-placement math behind <see cref="MuiTabShell"/> —
    /// unit-tested without a live Window. Mirrors
    /// `mui-tab-shell.component.ts`'s "top on tablet/desktop, bottom on
    /// phone" container-query breakpoint.
    /// </summary>
    public static class MuiTabShellMath
    {
        /// <summary>Below this host width, an Auto placement puts the tab
        /// bar at the bottom (thumb reach).</summary>
        public const double PhoneBreakpointPx = 640.0;

        public static bool TabBarAtBottom(MuiTabShellPlacement placement, double hostWidth) => placement switch
        {
            MuiTabShellPlacement.Bottom => true,
            MuiTabShellPlacement.Top => false,
            _ => hostWidth < PhoneBreakpointPx,
        };
    }
}

File: src/apple/Packages/MapleUI/Sources/MapleUI/Internal/MuiTabShellMath.swift
// MuiTabShellMath.swift — pure placement-resolution math for MuiTabShell
// (Templates, unified-component-catalog.md §5). "auto" resolves by host
// width rather than a hardware size class: the web reference's container
// query switches to a bottom tab bar below 640px, and on Apple platforms a
// width that narrow only ever happens on a phone in portrait, so a single
// width check reproduces the "top on tablet/desktop, bottom on phone"
// platform idiom without a separate size-class code path.

import Foundation

enum MuiTabShellMath {
    /// Matches the web reference's tab-shell container-query breakpoint.
    static let bottomBelowWidthPx: Double = 640

    /// Whether the tab bar renders at the bottom edge for `placement` at
    /// the given host width.
    static func isBottom(placement: MuiTabShellPlacement, hostWidth: Double) -> Bool {
        switch placement {
        case .top: return false
        case .bottom: return true
        case .auto: return hostWidth < bottomBelowWidthPx
        }
    }
}
